app/javascript/components/download/FileDownloadControl.js

```diff
@@ -0,0 +1,41 @@
+import React from 'react'
+import { fetchBucketFile } from 'lib/scp-api'
+import { bytesToSize } from 'lib/stats'
+
+/** renders a file download control for the given file object */
+export default function FileDownloadControl({ file, bucketName }) {
+  const fileName = file.upload_file_name
+
+  /** Load file when button is clicked */
+  const handleOnClick = () => {
```

Comment:
I think it's more conventional to name methods like this either `handleClick` or `handleDownloadClick` 
```suggestion
  const handleDownloadClick = () => {
```